fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/UserRolesSyncer.java

```diff
@@ -62,57 +76,78 @@ public void sync() {
   }
 
   public long syncAndReturn() {
-    updateServiceAccounts();
-    return updateUserPermissions();
-  }
-
-  private void updateServiceAccounts() {
-    AtomicInteger updateCount = new AtomicInteger(0);
-
     FixedBackOff backoff = new FixedBackOff();
     backoff.setInterval(retryIntervalMs);
     BackOffExecution backOffExec = backoff.start();
 
+    if (!isServerHealthy()) {
+      log.warn("Server is currently UNHEALTHY. User permission role synchronization and " +
+                   "resolution may not complete until this server becomes healthy again.");
```

Comment:
It won't mean anything for requests - but when used behind a load balancer, traffic will halt once the LB detects the instance has gone unhealthy. That has been the case ever since I introduced the `ResourceProviderHealthIndicator`, but now it's surfaced more in the logs for debugging.